flocker/node/agents/functional/test_ebs.py

```diff
@@ -69,15 +65,19 @@ def test_foreign_volume(self):
         Test that ``list_volumes`` lists only those volumes
         belonging to the current Flocker cluster.
         """
-        ec2_client = ec2_client_from_environment()
+        try:
+            cls, kwargs = get_blockdeviceapi_args(ProviderType.aws)
+        except ConfigMissing as e:
+            raise SkipTest(str(e))
+        ec2_client = kwargs["ec2_client"]
```

Comment:
This seems a little inside-out. But probably not worth worrying about at this point in time.